src/Role/Type/ContextRole.php

```diff
@@ -0,0 +1,178 @@
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Library\Lti1p3Core\Role\Type;
+
+use OAT\Library\Lti1p3Core\Role\AbstractRole;
+
+/**
+ * @see http://www.imsglobal.org/spec/lti/v1p3/#lis-vocabulary-for-context-roles
+ */
+class ContextRole extends AbstractRole
+{
+    public static function getType(): string
+    {
+        return static::TYPE_CONTEXT;
+    }
+
+    public static function getNameSpace(): string
+    {
+        return static::NAMESPACE_CONTEXT;
+    }
+
+    public function getSubName(): ?string
+    {
+        if (strpos($this->name, static::getNamespace() . '/') === 0) {
+            $exp = explode('#', $this->name);
+
+            if (sizeof($exp) === 2) {
+                return end($exp);
+            }
+        }
+
+        return null;
+    }
+
+    public function isCore(): bool
+    {
+        if (strpos($this->name, static::getNamespace() . '/') === 0) {
+            $exp = explode('#', substr($this->name, strlen(static::getNamespace() . '/')));
+
+            return $this->getMap()[current($exp)];
+        }
+
+        if (strpos($this->name, static::getNamespace()) === 0) {
+            $exp = explode('#', $this->name);
+
+            return $this->getMap()[end($exp)];
+        }
+
+        return $this->getMap()[$this->name];
+    }
+
+    protected function isValid(): bool
+    {
+        if (strpos($this->name, static::getNamespace() . '/') === 0) {
+            $exp = explode('#', substr($this->name, strlen(static::getNamespace() . '/')));
+
+            $mainName = current($exp);
+            $subName = end($exp);
+
+            return array_key_exists($mainName, $this->getMap()) && in_array($subName, $this->getSubMap()[$mainName]);
+        }
+
+        if (strpos($this->name, static::getNamespace()) === 0) {
+            $exp = explode('#', $this->name);
+
+            return array_key_exists(end($exp), $this->getMap());
+        }
+
+        return array_key_exists($this->name, $this->getMap());
+    }
+
+    protected function getMap(): array
+    {
+        return [
+            'Administrator' => true,
+            'ContentDeveloper' => true,
+            'Instructor' => true,
+            'Learner' => true,
+            'Mentor' => true,
+            'Manager' => false,
+            'Member' => false,
+            'Officer' => false,
+        ];
+    }
+
+    private function getSubMap(): array
+    {
+       return [
```

Comment:
Maybe it makes sense to put this structure to a YAML file?

Reply:
this is static  / frozen specs content, extracting this form the php class (owner) would make it's usage more complex and it requires anyway no updates

Reply:
the goal is not to work directly on those classes level, but on collection level to make RBAC easy to handle, see https://github.com/oat-sa/lib-lti1p3-core/blob/feature/validated-role-collection/doc/message/platform-originating-messages.md#working-with-launch-roles